tempodb/backend/mocks.go

```diff
@@ -72,16 +72,21 @@ func (m *MockRawReader) Shutdown() {}
 
 // MockRawWriter
 type MockRawWriter struct {
-	writeBuffer       []byte
+	writeBuffer       [][]byte
 	appendBuffer      []byte
 	closeAppendCalled bool
 	deleteCalls       map[string]map[string]int
 	err               error
 }
 
 func (m *MockRawWriter) Write(_ context.Context, _ string, _ KeyPath, data io.Reader, size int64, _ *CacheInfo) error {
-	var err error
-	m.writeBuffer, err = tempo_io.ReadAllWithEstimate(data, size)
+	if m.writeBuffer == nil {
+		m.writeBuffer = make([][]byte, 0)
+	}
+
+	writeBuffer, err := tempo_io.ReadAllWithEstimate(data, size)
+	m.writeBuffer = append(m.writeBuffer, writeBuffer)
```

Comment:
An alternative here would be to use the KeyPath and put the byte slices into a map. Probably about the same complexity as the `m.writeBuffer[len(m.writeBuffer)-2]` checks in the tests below, but thought I'd mention it as something to consider.